src/acl/RegexData.cc

```diff
@@ -108,13 +108,13 @@ removeUnnecessaryWildcards(char * t)
 }
 
 static void
-compileRE(std::list<RegexPattern> &curlist, const SBuf &RE, int flags)
+compileRE(std::list<RegexPattern> &curlist, const SBuf &RE, const std::regex::flag_type flags)
 {
     curlist.emplace_back(RE, flags);
```

Comment:
This whole configuration code (including, but not just compileOptimisedREs()) is clearly written to prioritize runtime performance over configuration-time overheads (and I agree with that tradeoff). AFAICT, we should add `std::regex::optimize` to the flags. If we should not, it would be a good idea to explain (via a source code comment) why not. 

```suggestion
    curlist.emplace_back(RE, flags | std::regex::optimize);
```

Alternatively, this extra flag would go into the RegexPattern constructor itself. We can probably make the same tradeoff claim for all RegexPattern users!

Reply:
Done in [b66b827](https://github.com/squid-cache/squid/pull/1232/commits/b66b827af33102ee75dbd7d4c8ce114fb396500f)